### ISSUE TITLE
[Research] Implement FEMBA EEG Foundation Model (7.8M params, edge-deployable)

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#24]
## What

Integrate **FEMBA** (IEEE EMBC 2025) — a bidirectional Mamba foundation model for EEG with a **7.8M-parameter tiny variant** suitable for edge/wearable deployment. Pretrained on 21,000+ hours of unlabeled EEG, achieves 81.82% balanced accuracy on TUAB.

## Why

- Lighter alternative to REVE (issue #19) — only 7.8M params vs REVE's larger footprint
- **Linear-time complexity** (Mamba architecture) — ideal for real-time 4-second epoch processing
- Pretrained on 21K hours of EEG — has learned general EEG representations
- Weights available on HuggingFace: `PulpBio/FEMBA`
- Could potentially run in-browser via ONNX/WebAssembly (edge-AI privacy)
- Bidirectional processing captures both forward and backward temporal context

### Step 1: Download FEMBA weights
```bash
pip install mamba-ssm  # or equivalent
python -c "from huggingface_hub import snapshot_download; snapshot_download('PulpBio/FEMBA', local_dir='ml/models/saved/femba')"
```

### Step 2: Create Muse 2 adapter
```python
# ml/models/femba_adapter.py
class FEMBAEmotionClassifier:
    """FEMBA foundation model adapted for Muse 2 emotion classification.
    
    Architecture:
    - Tokenization: 2D conv + flatten (from raw EEG epochs)
    - Encoder: bidirectional Mamba blocks (hidden_size=80)
    - Classifier: linear head or Mamba-enhanced head (0.5-0.7M additional params)
    
    Input: (4_channels, 1024_samples) = 4 seconds at 256 Hz
    Output: 6-class emotion probabilities + valence + arousal
    """
    def __init__(self):
        self.model = load_femba_tiny("ml/models/saved/femba")
        self.classifier = nn.Linear(80, 6)  # 6 emotions
        
    def fine_tune(self, eeg_dataset, labels):
        """Fine-tune classifier head on emotion data.
        Stage 1: freeze encoder, train linear head
        Stage 2: unfreeze, fine-tune end-to-end
        """
```

### Step 3: Channel adaptation
- FEMBA pretrained on multi-channel EEG (likely 19+ channels)
- For Muse 2: use only 4 channels, adapt input layer
- Consider: pretrain on 4-channel subsets of SEED/DEAP to bridge the gap

### Step 4: ONNX export for edge deployment
```python
# Export fine-tuned model to ONNX for browser inference
torch.onnx.export(model, dummy_input, "femba_emotion.onnx")
# Convert to ONNX Runtime Web for in-browser inference
```

### Files to Modify
- `ml/models/femba_adapter.py` (NEW) — FEMBA wrapper for emotion
- `ml/models/emotion_classifier.py` — Add FEMBA to prediction chain
- `ml/training/train_femba_emotion.py` (NEW) — Fine-tuning script
- `ml/requirements.txt` — Add mamba-ssm, torch

### Expected Impact
- EEG emotion accuracy: **74% → 82-86%** (estimated, with fine-tuning)
- Inference latency: <50ms per 4-second epoch (linear complexity)
- Future: browser-side inference (privacy-preserving, no server needed)

## Complexity
MEDIUM — pretrained weights available, smaller model than REVE, but Mamba dependency

## References
- FEMBA paper: https://arxiv.org/abs/2502.06438
- HuggingFace weights: https://huggingface.co/PulpBio/FEMBA
- Architecture: bidirectional Mamba, 7.8M params, 80-dim hidden
- Performance: 81.82% balanced accuracy (TUAB), 0.949 AUROC (TUAR)
